src/sharing/SharingScreen.js

```diff
@@ -0,0 +1,498 @@
+              onFocus={this.focusStream}
+              onChange={this.focusStream}
+              onBlur={this.blurStream}
+              selectTextOnFocus
+            />
+            <View />
+            <AnimatedScaleComponent visible={isTopicFocused}>
+              <TopicAutocomplete
+                text={topic}
+                onAutocomplete={this.handleTopicAutoComplete}
+                isFocused={isTopicFocused}
+                narrow={narrow}
+              />
+            </AnimatedScaleComponent>
+            <Input
+              placeholder="Topic"
+              value={topic}
+              onFocus={this.focusTopic}
+              onBlur={this.blurTopic}
+              onChangeText={this.handleTopicChange}
+              editable={stream !== ''}
+            />
+            <Input
+              placeholder="Message"
+              value={message}
+              onChangeText={this.handleMessageChange}
+              multiline
+            />
+          </View>
+        </ScrollView>
+        <View style={styles.actions}>
+          <ZulipButton onPress={this.finishShare} style={styles.button} secondary text="Cancel" />
+          <ZulipButton
+            style={styles.button}
+            onPress={this.handleSend}
+            text="Send"
+            progress={sending}
+            disabled={!this.isSendButtonEnabled()}
+          />
+        </View>
+      </>
+    );
+  }
+}
+
+type ShareToPmProps = $ReadOnly<{|
+  ...$Exact<NavigationNavigatorProps<{||}, {| params: {| sharedData: SharedData |} |}>>,
+  dispatch: Dispatch,
+  auth: Auth,
+|}>;
+
+type ShareToPmState = $ReadOnly<{|
+  users: User[],
+  message: string,
+  choosingRecipients: boolean,
+  sending: boolean,
+|}>;
+
+class ShareToPmComponent extends React.Component<ShareToPmProps, ShareToPmState> {
+  constructor(props) {
+    super(props);
+    const { sharedData } = this.props.navigation.state.params;
+    this.state = {
+      users: [],
+      message: sharedData.type === 'text' ? sharedData.sharedText : '',
+      choosingRecipients: false,
+      sending: false,
+    };
+  }
+
+  setSending = () => {
+    this.setState({ sending: true });
+  };
+
+  toggleChoosingRecipients = () => {
+    this.setState(state => ({ choosingRecipients: !state.choosingRecipients }));
+  };
+
+  handleChooseRecipients = (users: Array<User>) => {
+    this.setState({ users });
+    this.toggleChoosingRecipients();
+  };
+
+  finishShare = () => {
+    const { dispatch } = this.props;
+
+    dispatch(navigateBack());
+    BackHandler.exitApp();
+  };
+
+  handleMessageChange = message => {
+    this.setState({ message });
+  };
+
+  isSendButtonEnabled = () => {
+    const { message, users } = this.state;
+    const { sharedData } = this.props.navigation.state.params;
+
+    if (sharedData.type === 'text') {
+      return message !== '' && users.length > 0;
+    }
+
+    return users.length > 0;
+  };
+
+  getUsersPreview = () => {
```

Comment:
This is basically a render method, a helper for the component's main render method. So let's put "render" in the name, like `renderUsersPreview`.

Reply:
Changed to `renderUsersPreview`.